rfc/0006-kubewarden-versioning.md

```diff
@@ -0,0 +1,203 @@
+|              |                                 |
+|:-------------|:--------------------------------|
+| Feature Name | Kubewarden versioning           |
+| Start Date   | Mar 3rd 2022                    |
+| Category     | Versioning                      |
+| RFC PR       | [fill this in after opening PR] |
+
+# Summary
+[summary]: #summary
+
+Kubewarden is composed of many core components that work together in
+order to build the final product. This is a proposal to harmonize
+Kubewarden versioning schema.
+
+# Motivation
+[motivation]: #motivation
+
+Having a way to specify what version of Kubewarden is being installed
+or executed would make it easier for the Kubewarden team to describe,
+market and support Kubewarden (as in user support).
+
+It would help users in understanding what version of Kubewarden they
+are about to install or execute.
+
+## Examples / User Stories
+[examples]: #examples
+
+> Users do not need to keep track of component versions and how they
+> work together. They just need to care about running a specific
+> Kubewarden version.
+
+> Users are able to easily understand that a new version of Kubewarden
+> is available, and that more than one component might be upgraded in
+> order to move to the new Kubewarden version.
+
+> A user has an issue with Kubewarden. Given its multiple component
+> nature, we would like to know what was the Kubewarden version that
+> got installed, and through what versions it was upgraded.
+
+# Detailed design
+[design]: #detailed-design
+
+## Many components, one experience
+
+Kubewarden core components, which are exposed to the user are:
+
+- Helm Charts
+- Kubewarden Controller
+- Policy Server
+- kwctl
+
+As a second dimension, there are also policies and their respective
+SDK's. This proposal is going to ignore this dimension in purpose,
+because Kubewarden runtime components (`policy-server` and `kwctl`)
+should be compatible with policies previously built, even with older
+language SDK's.
+
+### Public interfaces
+
+Our public interfaces with users are as follows:
```

Comment:
The use of 'users' in this context might be too generic, as we have
- policy consumers (aka end users, who try to deploy something on k8s and could fail due to a policy)
- cluster admins (those who define which policies are enforced)
- policy authors